lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -271,25 +264,40 @@ def valid_event(event_data: dict) -> bool:
     return all((record.get("body") is not None) for record in event_data["Records"])
 
 
-def forward_to_service(inbound_sms: dict, url: str) -> bool:
+def forward_to_service(inbound_sms: dict, url: str, auth_parameter: str) -> bool:
     """
     Forwards the inbound SMS to the service that has 2-way SMS setup.
     """
 
+    global client_auth_token
+
+    if client_auth_token != 'test':
+        try:
+            client_auth_token = get_ssm_param_info(client_api_auth_ssm_path=auth_parameter)
+        except Exception as e:
+            logger.exception("Issue attempting to get ssm parameter for incoming two-way sms - Exception: %s", e)
+
+        if client_auth_token is None:
+            logger.critical(
+                'Raising execption because client_auth_token could not be retrieved from SSM. client_auth_token = %s',
+                client_auth_token
+            )
+            raise Exception('Raising execption because client_auth_token could not be retrieved from SSM.')
+
     # This covers None and the empty string.
     if not url:
         logger.error("No URL provided in the configuration for the service.")
         return False
 
     headers = {
         'Content-type': 'application/json',
-        'Authorization': 'Basic ' + vetext_auth_token
+        'Authorization': 'Basic ' + client_auth_token
     }
 
     try:
         response = requests.post(
-            url,            
-            verify=False,
+            url,
+            verify=False if auth_parameter and 'vetext' in auth_parameter else True,
```

Comment:
I don't believe we can make it down here with `auth_parameter` being None or an empty string.

Reply:
I was hitting an issue there during testing. Maybe the tests just need reworked a little bit.

Reply:
adjusted unit tests to send a str instead of `None` type, and removed extra `auth_parameter` check